cachi2/core/package_managers/rubygems/main.py

```diff
@@ -1,7 +1,99 @@
+import logging
+from typing import Optional
+
 from cachi2.core.models.input import Request
-from cachi2.core.models.output import RequestOutput
+from cachi2.core.models.output import EnvironmentVariable, ProjectFile, RequestOutput
+from cachi2.core.models.sbom import Component
+from cachi2.core.package_managers.rubygems.parser import (
+    BundlerDependency,
+    GemDependency,
+    GitDependency,
+    PathDependency,
+    parse_gemlock,
+)
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(__name__)
 
 
 def fetch_rubygems_source(request: Request) -> RequestOutput:
     """Resolve and process all rubygems packages."""
+    components: list[Component] = []
+    environment_variables: list[EnvironmentVariable] = []
+    project_files: list[ProjectFile] = []
+
+    for package in request.packages:
+        package_dir = request.source_dir.join_within_root(package.path)
+        _resolve_rubygems(package_dir=package_dir, output_dir=request.output_dir)
+
+    return RequestOutput.from_obj_list(
+        components=components,
+        environment_variables=environment_variables,
+        project_files=project_files,
+    )
+
+
+def _resolve_rubygems(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
+    """Process a request for a single rubygems package."""
+    dependencies = parse_gemlock(package_dir)
+    _download_dependencies(output_dir, dependencies)
+    return _create_components(package_dir, dependencies)
+
+
+def _create_components(
+    package_dir: RootedPath,
+    dependencies: list[BundlerDependency],
+) -> list[Component]:
+    """TODO."""
+    return NotImplemented
+
+
+def _generate_purl_for(gem: BundlerDependency) -> str:
```

Comment:
this one should likely go to a `@dataclass/NamedTuple` instance representing a package, kinda like it's done in gomod and npm and now also proposed for rpm (#600)